domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -1527,7 +1547,16 @@ QueryHelper buildSelectQueryHelper(
 
     QueryHelper queryHelper = new QueryHelper(transactionMode, table, facets);
 
-    queryHelper.selectQuery = SELECT + fieldName + addIdField + FROM + schemaName + DOT + table + SPACE + where;
+    String distinctOnClause = "";
+    if (distinctOn != null && !distinctOn.isEmpty()) {
+      String[] fields = distinctOn.split(",");
+      String distinctOnFields = Stream.of(fields)
+          .map(str -> String.format("%s", str))
+          .collect(Collectors.joining(","));
+      distinctOnClause = String.format("DISTINCT ON (%s) ", distinctOnFields);
```

Comment:
What does `str -> String.format("%s", str)` do?
Isn't this sufficient?

    if (distinctOn != null && !distinctOn.isEmpty()) {
      distinctOnClause = "DISTINCT ON (" + distinctOn + ")";
    }

Reply:
Right, this code is redundant. So it simplified now.